subsys/shell/shell.c

```diff
@@ -45,6 +45,21 @@ static int shell_execute(const struct shell *shell);
 extern const struct shell_cmd_entry __shell_root_cmds_start[0];
 extern const struct shell_cmd_entry __shell_root_cmds_end[0];
 
+static inline bool ignore_nl(const struct shell *shell, uint8_t new_line)
+{
+	if ((shell->ctx->internal.flags.last_nl != '\r') &&
```

Comment:
don't you need to assign every byte to last_nl to detect consecutive \r and \n?

Reply:
It is enough to save last state once, when commented condition is met.
Last new line character is reset when any character other than NL is received.